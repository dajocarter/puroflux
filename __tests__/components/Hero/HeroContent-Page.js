import React from 'react'
import renderer from 'react-test-renderer'

import HeroContentPage from '../../../src/components/Hero/HeroContent-Page'

const sampleHTML = '<p>Some HTML</p>'
const sampleButtons = [
  {
    button_link: {
      target: '_blank',
      title: 'External Link',
      url: 'https://helloworld.com'
    }
  },
  {
    button_link: {
      target: '',
      title: 'Local Link',
      url: '/page'
    }
  }
]

describe('Hero Content', () => {
  it('renders nothing without props', () => {
    const tree = renderer
      .create(<HeroContentPage />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders HTML', () => {
    const tree = renderer
      .create(<HeroContentPage html={sampleHTML} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders buttons', () => {
    const tree = renderer
      .create(<HeroContentPage buttons={sampleButtons} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders both HTML and buttons', () => {
    const tree = renderer
      .create(<HeroContentPage html={sampleHTML} buttons={sampleButtons} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
