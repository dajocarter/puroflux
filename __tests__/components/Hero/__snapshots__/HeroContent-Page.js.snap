// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Content renders HTML 1`] = `
<div
  className="content"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Some HTML</p>",
    }
  }
/>
`;

exports[`Hero Content renders both HTML and buttons 1`] = `
Array [
  <div
    className="content"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Some HTML</p>",
      }
    }
  />,
  <div
    className="sc-bwzfXH iqaXvV"
  >
    <a
      className="sc-bdVaJa sc-htpNat jEQdjd"
      href="https://helloworld.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      External Link
    </a>
    <a
      className="sc-bdVaJa sc-htpNat jEQdjd"
      href="/page"
      primary="false"
      secondary="true"
    >
      Local Link
    </a>
  </div>,
]
`;

exports[`Hero Content renders buttons 1`] = `
<div
  className="sc-bwzfXH iqaXvV"
>
  <a
    className="sc-bdVaJa sc-htpNat jEQdjd"
    href="https://helloworld.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    External Link
  </a>
  <a
    className="sc-bdVaJa sc-htpNat jEQdjd"
    href="/page"
    primary="false"
    secondary="true"
  >
    Local Link
  </a>
</div>
`;

exports[`Hero Content renders nothing without props 1`] = `null`;
